{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1448A-WT-KB-1_combined',\n",
       " '1448A-WT-KB-2_combined',\n",
       " '1448A-WT-MM-1_combined',\n",
       " '1448A-WT-MM-2_combined',\n",
       " 'PSPH-WT-KB-1_combined',\n",
       " 'PSPH-WT-KB-2_combined',\n",
       " 'PSPH-WT-MM-1_combined',\n",
       " 'PSPH-WT-MM-2_combined',\n",
       " 'S0858-KB-1_combined',\n",
       " 'S0858-KB-2_combined',\n",
       " 'S0858-MM-1_combined',\n",
       " 'S0858-MM-2_combined',\n",
       " 'Vfr-KB-1_combined',\n",
       " 'Vfr-KB-2_combined',\n",
       " 'Vfr-MM-1_combined',\n",
       " 'Vfr-MM-2_combined',\n",
       " 'aefR-KB-1_combined',\n",
       " 'aefR-KB-2_combined',\n",
       " 'aefR-MM-1_combined',\n",
       " 'aefR-MM-2_combined',\n",
       " 'cvsS-KB-1_combined',\n",
       " 'cvsS-KB-2_combined',\n",
       " 'cvsS-MM-1_combined',\n",
       " 'cvsS-MM-2_combined',\n",
       " 'envZ-KB-1_combined',\n",
       " 'envZ-KB-2_combined',\n",
       " 'envZ-MM-1_combined',\n",
       " 'envZ-MM-2_combined',\n",
       " 'gacA-KB-1_combined',\n",
       " 'gacA-KB-2_combined',\n",
       " 'gacA-MM-1_combined',\n",
       " 'gacA-MM-2_combined',\n",
       " 'gtrS-KB-1_combined',\n",
       " 'gtrS-KB-2_combined',\n",
       " 'gtrS-MM-1_combined',\n",
       " 'gtrS-MM-2_combined',\n",
       " 'ompR-KB-1_combined',\n",
       " 'ompR-KB-2_combined',\n",
       " 'ompR-MM-1_combined',\n",
       " 'ompR-MM-2_combined',\n",
       " 'ox-rpoN-KB-1_combined',\n",
       " 'ox-rpoN-KB-2_combined',\n",
       " 'ox-rpoN-MM-1_combined',\n",
       " 'ox-rpoN-MM-2_combined',\n",
       " 'pilR-KB-1_combined',\n",
       " 'pilR-KB-2_combined',\n",
       " 'pilR-MM-1_combined',\n",
       " 'pilR-MM-2_combined',\n",
       " 'pilS-KB-1_combined',\n",
       " 'pilS-KB-2_combined',\n",
       " 'pilS-MM-1_combined',\n",
       " 'pilS-MM-2_combined',\n",
       " 'psrA-KB-1_combined',\n",
       " 'psrA-KB-2_combined',\n",
       " 'psrA-MM-1_combined',\n",
       " 'psrA-MM-2_combined']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get all sample name\n",
    "\n",
    "import glob\n",
    "raw_files = glob.glob('01_rawdata/*.gz')\n",
    "raw_files\n",
    "sample = [i.split('/')[1] for i in raw_files] \n",
    "sample = [i.split('_R1.fast')[0] for i in sample] \n",
    "sample = [i.split('_R2.fast')[0] for i in sample] \n",
    "sample = list(set(sample))\n",
    "sample.sort()\n",
    "sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# curate samples we used\n",
    "\n",
    "sample = ['1448A-WT-KB-1_combined','1448A-WT-KB-2_combined','1448A-WT-MM-1_combined','1448A-WT-MM-2_combined',\n",
    " 'PSPH-WT-KB-1_combined','PSPH-WT-KB-2_combined','PSPH-WT-MM-1_combined','PSPH-WT-MM-2_combined',\n",
    " 'S0858-KB-1_combined','S0858-KB-2_combined','S0858-MM-1_combined','S0858-MM-2_combined',\n",
    " 'Vfr-KB-1_combined','Vfr-KB-2_combined','Vfr-MM-1_combined','Vfr-MM-2_combined',\n",
    " 'aefR-KB-1_combined','aefR-KB-2_combined','aefR-MM-1_combined','aefR-MM-2_combined',\n",
    " 'cvsS-KB-1_combined','cvsS-KB-2_combined','cvsS-MM-1_combined','cvsS-MM-2_combined',\n",
    " 'envZ-KB-1_combined','envZ-KB-2_combined','envZ-MM-1_combined','envZ-MM-2_combined',\n",
    " 'gacA-KB-1_combined','gacA-KB-2_combined','gacA-MM-1_combined','gacA-MM-2_combined',\n",
    " 'gtrS-KB-1_combined','gtrS-KB-2_combined','gtrS-MM-1_combined','gtrS-MM-2_combined',\n",
    " 'ompR-KB-1_combined','ompR-KB-2_combined','ompR-MM-1_combined','ompR-MM-2_combined',\n",
    " 'ox-rpoN-KB-1_combined','ox-rpoN-KB-2_combined','ox-rpoN-MM-1_combined','ox-rpoN-MM-2_combined',\n",
    " 'pilR-KB-1_combined','pilR-KB-2_combined','pilR-MM-1_combined','pilR-MM-2_combined',\n",
    " 'pilS-KB-1_combined','pilS-KB-2_combined','pilS-MM-1_combined','pilS-MM-2_combined',\n",
    " 'psrA-KB-1_combined','psrA-KB-2_combined','psrA-MM-1_combined','psrA-MM-2_combined']\n",
    "\n",
    "thread = 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# mutant gene\n",
    "\n",
    "genename = ['hrpL','hrpR','hrpS','rhpR',\n",
    "            'phoP','mgrA','PSPPH_0857','pilR',\n",
    "            'ompR','gltR','vfr','algU',\n",
    "            'CvsR','GacA','PsrA','aefR']\n",
    "genelist = ['gene-PSPPH_RS06545','gene-PSPPH_RS06420','gene-PSPPH_RS06425','PSPPH_3730',\n",
    "            'PSPPH_RS10105','gene-PSPPH_3645','gene-PSPPH_RS04395','gene-PSPPH_RS03755',\n",
    "            'gene-PSPPH_RS01270','gene-PSPPH_RS05985','gene-PSPPH_RS03440','gene-PSPPH_RS19995',\n",
    "            'PSPPH_RS15855','gene-PSPPH_RS11795','gene-PSPPH_RS16230','PSPPH_RS16430']\n",
    "gene = ['PSPPH_1294','PSPPH_1270','PSPPH_1271','PSPPH_2004',\n",
    "        'PSPPH_3730','PSPPH_3645','PSPPH_0857','PSPPH_0736',\n",
    "        'PSPPH_0246','PSPPH_1179','PSPPH_0676','PSPPH_3955',\n",
    "        'PSPPH_3127','PSPPH_2328','PSPPH_3202','PSPPH_3244']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "110"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sample)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir 01_fastqc\n",
      "mkdir 02_trim\n",
      "mkdir 03_star\n",
      "mkdir 03_tpm\n",
      "nohup bash 01_fastqc/01_fastqc.sh > 01_fastqc/01_fastqc.log 2>&1 &\n",
      "nohup bash 02_trim/02_trim.sh > 02_trim/02_trim.log 2>&1 &\n",
      "nohup bash 03_star/03_star.sh > 03_star/03_star.log 2>&1 &\n",
      "nohup bash 03_tpm/03_tpm.sh > 03_tpm/03_tpm.log 2>&1 &\n"
     ]
    }
   ],
   "source": [
    "# mk main output dir\n",
    "main_dir = ['01_fastqc','02_trim','03_star','03_tpm']\n",
    "    \n",
    "for i in main_dir:\n",
    "    print('mkdir %s'%i)\n",
    "    \n",
    "for i in main_dir:\n",
    "    print('nohup bash %s/%s.sh > %s/%s.log 2>&1 &'%(i,i,i,i))\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fastqc -f fastq -t 20 -o 01_fastqc 01_rawdata/*/*.fastq.gz\n",
      "echo '************************************** fastqc finished! **************************************'\n"
     ]
    }
   ],
   "source": [
    "# fastqc\n",
    "print('fastqc -f fastq -t %s -o 01_fastqc 01_rawdata/*.fastq.gz'%(thread))\n",
    "\n",
    "print('echo \\'************************************** fastqc finished! **************************************\\'')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir 02_trim/1448A-WT-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/1448A-WT-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/1448A-WT-KB-1_combined*R1* 01_rawdata/1448A-WT-KB-1_combined*R2*\n",
      "mkdir 02_trim/1448A-WT-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/1448A-WT-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/1448A-WT-KB-2_combined*R1* 01_rawdata/1448A-WT-KB-2_combined*R2*\n",
      "mkdir 02_trim/1448A-WT-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/1448A-WT-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/1448A-WT-MM-1_combined*R1* 01_rawdata/1448A-WT-MM-1_combined*R2*\n",
      "mkdir 02_trim/1448A-WT-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/1448A-WT-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/1448A-WT-MM-2_combined*R1* 01_rawdata/1448A-WT-MM-2_combined*R2*\n",
      "mkdir 02_trim/PSPH-WT-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/PSPH-WT-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/PSPH-WT-KB-1_combined*R1* 01_rawdata/PSPH-WT-KB-1_combined*R2*\n",
      "mkdir 02_trim/PSPH-WT-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/PSPH-WT-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/PSPH-WT-KB-2_combined*R1* 01_rawdata/PSPH-WT-KB-2_combined*R2*\n",
      "mkdir 02_trim/PSPH-WT-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/PSPH-WT-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/PSPH-WT-MM-1_combined*R1* 01_rawdata/PSPH-WT-MM-1_combined*R2*\n",
      "mkdir 02_trim/PSPH-WT-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/PSPH-WT-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/PSPH-WT-MM-2_combined*R1* 01_rawdata/PSPH-WT-MM-2_combined*R2*\n",
      "mkdir 02_trim/S0858-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/S0858-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/S0858-KB-1_combined*R1* 01_rawdata/S0858-KB-1_combined*R2*\n",
      "mkdir 02_trim/S0858-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/S0858-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/S0858-KB-2_combined*R1* 01_rawdata/S0858-KB-2_combined*R2*\n",
      "mkdir 02_trim/S0858-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/S0858-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/S0858-MM-1_combined*R1* 01_rawdata/S0858-MM-1_combined*R2*\n",
      "mkdir 02_trim/S0858-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/S0858-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/S0858-MM-2_combined*R1* 01_rawdata/S0858-MM-2_combined*R2*\n",
      "mkdir 02_trim/Vfr-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/Vfr-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/Vfr-KB-1_combined*R1* 01_rawdata/Vfr-KB-1_combined*R2*\n",
      "mkdir 02_trim/Vfr-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/Vfr-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/Vfr-KB-2_combined*R1* 01_rawdata/Vfr-KB-2_combined*R2*\n",
      "mkdir 02_trim/Vfr-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/Vfr-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/Vfr-MM-1_combined*R1* 01_rawdata/Vfr-MM-1_combined*R2*\n",
      "mkdir 02_trim/Vfr-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/Vfr-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/Vfr-MM-2_combined*R1* 01_rawdata/Vfr-MM-2_combined*R2*\n",
      "mkdir 02_trim/aefR-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/aefR-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/aefR-KB-1_combined*R1* 01_rawdata/aefR-KB-1_combined*R2*\n",
      "mkdir 02_trim/aefR-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/aefR-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/aefR-KB-2_combined*R1* 01_rawdata/aefR-KB-2_combined*R2*\n",
      "mkdir 02_trim/aefR-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/aefR-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/aefR-MM-1_combined*R1* 01_rawdata/aefR-MM-1_combined*R2*\n",
      "mkdir 02_trim/aefR-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/aefR-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/aefR-MM-2_combined*R1* 01_rawdata/aefR-MM-2_combined*R2*\n",
      "mkdir 02_trim/cvsS-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/cvsS-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/cvsS-KB-1_combined*R1* 01_rawdata/cvsS-KB-1_combined*R2*\n",
      "mkdir 02_trim/cvsS-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/cvsS-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/cvsS-KB-2_combined*R1* 01_rawdata/cvsS-KB-2_combined*R2*\n",
      "mkdir 02_trim/cvsS-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/cvsS-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/cvsS-MM-1_combined*R1* 01_rawdata/cvsS-MM-1_combined*R2*\n",
      "mkdir 02_trim/cvsS-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/cvsS-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/cvsS-MM-2_combined*R1* 01_rawdata/cvsS-MM-2_combined*R2*\n",
      "mkdir 02_trim/envZ-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/envZ-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/envZ-KB-1_combined*R1* 01_rawdata/envZ-KB-1_combined*R2*\n",
      "mkdir 02_trim/envZ-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/envZ-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/envZ-KB-2_combined*R1* 01_rawdata/envZ-KB-2_combined*R2*\n",
      "mkdir 02_trim/envZ-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/envZ-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/envZ-MM-1_combined*R1* 01_rawdata/envZ-MM-1_combined*R2*\n",
      "mkdir 02_trim/envZ-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/envZ-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/envZ-MM-2_combined*R1* 01_rawdata/envZ-MM-2_combined*R2*\n",
      "mkdir 02_trim/gacA-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gacA-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gacA-KB-1_combined*R1* 01_rawdata/gacA-KB-1_combined*R2*\n",
      "mkdir 02_trim/gacA-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gacA-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gacA-KB-2_combined*R1* 01_rawdata/gacA-KB-2_combined*R2*\n",
      "mkdir 02_trim/gacA-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gacA-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gacA-MM-1_combined*R1* 01_rawdata/gacA-MM-1_combined*R2*\n",
      "mkdir 02_trim/gacA-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gacA-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gacA-MM-2_combined*R1* 01_rawdata/gacA-MM-2_combined*R2*\n",
      "mkdir 02_trim/gtrS-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gtrS-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gtrS-KB-1_combined*R1* 01_rawdata/gtrS-KB-1_combined*R2*\n",
      "mkdir 02_trim/gtrS-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gtrS-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gtrS-KB-2_combined*R1* 01_rawdata/gtrS-KB-2_combined*R2*\n",
      "mkdir 02_trim/gtrS-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gtrS-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gtrS-MM-1_combined*R1* 01_rawdata/gtrS-MM-1_combined*R2*\n",
      "mkdir 02_trim/gtrS-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/gtrS-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/gtrS-MM-2_combined*R1* 01_rawdata/gtrS-MM-2_combined*R2*\n",
      "mkdir 02_trim/ompR-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ompR-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ompR-KB-1_combined*R1* 01_rawdata/ompR-KB-1_combined*R2*\n",
      "mkdir 02_trim/ompR-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ompR-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ompR-KB-2_combined*R1* 01_rawdata/ompR-KB-2_combined*R2*\n",
      "mkdir 02_trim/ompR-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ompR-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ompR-MM-1_combined*R1* 01_rawdata/ompR-MM-1_combined*R2*\n",
      "mkdir 02_trim/ompR-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ompR-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ompR-MM-2_combined*R1* 01_rawdata/ompR-MM-2_combined*R2*\n",
      "mkdir 02_trim/ox-rpoN-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ox-rpoN-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ox-rpoN-KB-1_combined*R1* 01_rawdata/ox-rpoN-KB-1_combined*R2*\n",
      "mkdir 02_trim/ox-rpoN-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ox-rpoN-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ox-rpoN-KB-2_combined*R1* 01_rawdata/ox-rpoN-KB-2_combined*R2*\n",
      "mkdir 02_trim/ox-rpoN-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ox-rpoN-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ox-rpoN-MM-1_combined*R1* 01_rawdata/ox-rpoN-MM-1_combined*R2*\n",
      "mkdir 02_trim/ox-rpoN-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/ox-rpoN-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/ox-rpoN-MM-2_combined*R1* 01_rawdata/ox-rpoN-MM-2_combined*R2*\n",
      "mkdir 02_trim/pilR-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilR-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilR-KB-1_combined*R1* 01_rawdata/pilR-KB-1_combined*R2*\n",
      "mkdir 02_trim/pilR-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilR-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilR-KB-2_combined*R1* 01_rawdata/pilR-KB-2_combined*R2*\n",
      "mkdir 02_trim/pilR-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilR-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilR-MM-1_combined*R1* 01_rawdata/pilR-MM-1_combined*R2*\n",
      "mkdir 02_trim/pilR-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilR-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilR-MM-2_combined*R1* 01_rawdata/pilR-MM-2_combined*R2*\n",
      "mkdir 02_trim/pilS-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilS-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilS-KB-1_combined*R1* 01_rawdata/pilS-KB-1_combined*R2*\n",
      "mkdir 02_trim/pilS-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilS-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilS-KB-2_combined*R1* 01_rawdata/pilS-KB-2_combined*R2*\n",
      "mkdir 02_trim/pilS-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilS-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilS-MM-1_combined*R1* 01_rawdata/pilS-MM-1_combined*R2*\n",
      "mkdir 02_trim/pilS-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/pilS-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/pilS-MM-2_combined*R1* 01_rawdata/pilS-MM-2_combined*R2*\n",
      "mkdir 02_trim/psrA-KB-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/psrA-KB-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/psrA-KB-1_combined*R1* 01_rawdata/psrA-KB-1_combined*R2*\n",
      "mkdir 02_trim/psrA-KB-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/psrA-KB-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/psrA-KB-2_combined*R1* 01_rawdata/psrA-KB-2_combined*R2*\n",
      "mkdir 02_trim/psrA-MM-1_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/psrA-MM-1_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/psrA-MM-1_combined*R1* 01_rawdata/psrA-MM-1_combined*R2*\n",
      "mkdir 02_trim/psrA-MM-2_combined\n",
      "trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/psrA-MM-2_combined --paired -a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG 01_rawdata/psrA-MM-2_combined*R1* 01_rawdata/psrA-MM-2_combined*R2*\n",
      "echo '************************************** trim finished! **************************************'\n"
     ]
    }
   ],
   "source": [
    "# qc trim\n",
    "\n",
    "with open('02_trim/02_trim.sh', 'w') as f:\n",
    "    for i in sample:\n",
    "        print('mkdir 02_trim/%s'%i)\n",
    "        f.write('mkdir 02_trim/%s\\n'%i)\n",
    "        print('trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/%s --paired \\\n",
    "-a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG \\\n",
    "01_rawdata/%s*R1* 01_rawdata/%s*R2*'%(i,i,i))\n",
    "        f.write('trim_galore -q 20 --phred33 --length 36 --fastqc -o 02_trim/%s --paired \\\n",
    "-a AGATCGGAAGAGCACACGTCTGAACTCCAGTCA -a2 AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTG \\\n",
    "01_rawdata/%s*R1* 01_rawdata/%s*R2*\\n'%(i,i,i))\n",
    "    print('echo \\'************************************** trim finished! **************************************\\'')\n",
    "    f.write('echo \\'************************************** trim finished! **************************************\\'')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir 03_star/1448A-WT-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_R1_val_1.fq.gz 02_trim/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/1448A-WT-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_R1_val_1.fq.gz 02_trim/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/1448A-WT-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_R1_val_1.fq.gz 02_trim/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/1448A-WT-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_R1_val_1.fq.gz 02_trim/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/PSPH-WT-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_R1_val_1.fq.gz 02_trim/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/PSPH-WT-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_R1_val_1.fq.gz 02_trim/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/PSPH-WT-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_R1_val_1.fq.gz 02_trim/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/PSPH-WT-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_R1_val_1.fq.gz 02_trim/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/S0858-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/S0858-KB-1_combined/S0858-KB-1_combined_R1_val_1.fq.gz 02_trim/S0858-KB-1_combined/S0858-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/S0858-KB-1_combined/S0858-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/S0858-KB-1_combined/S0858-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/S0858-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/S0858-KB-2_combined/S0858-KB-2_combined_R1_val_1.fq.gz 02_trim/S0858-KB-2_combined/S0858-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/S0858-KB-2_combined/S0858-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/S0858-KB-2_combined/S0858-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/S0858-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/S0858-MM-1_combined/S0858-MM-1_combined_R1_val_1.fq.gz 02_trim/S0858-MM-1_combined/S0858-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/S0858-MM-1_combined/S0858-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/S0858-MM-1_combined/S0858-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/S0858-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/S0858-MM-2_combined/S0858-MM-2_combined_R1_val_1.fq.gz 02_trim/S0858-MM-2_combined/S0858-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/S0858-MM-2_combined/S0858-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/S0858-MM-2_combined/S0858-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/Vfr-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/Vfr-KB-1_combined/Vfr-KB-1_combined_R1_val_1.fq.gz 02_trim/Vfr-KB-1_combined/Vfr-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/Vfr-KB-1_combined/Vfr-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/Vfr-KB-1_combined/Vfr-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/Vfr-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/Vfr-KB-2_combined/Vfr-KB-2_combined_R1_val_1.fq.gz 02_trim/Vfr-KB-2_combined/Vfr-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/Vfr-KB-2_combined/Vfr-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/Vfr-KB-2_combined/Vfr-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/Vfr-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/Vfr-MM-1_combined/Vfr-MM-1_combined_R1_val_1.fq.gz 02_trim/Vfr-MM-1_combined/Vfr-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/Vfr-MM-1_combined/Vfr-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/Vfr-MM-1_combined/Vfr-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/Vfr-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/Vfr-MM-2_combined/Vfr-MM-2_combined_R1_val_1.fq.gz 02_trim/Vfr-MM-2_combined/Vfr-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/Vfr-MM-2_combined/Vfr-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/Vfr-MM-2_combined/Vfr-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/aefR-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/aefR-KB-1_combined/aefR-KB-1_combined_R1_val_1.fq.gz 02_trim/aefR-KB-1_combined/aefR-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/aefR-KB-1_combined/aefR-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/aefR-KB-1_combined/aefR-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/aefR-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/aefR-KB-2_combined/aefR-KB-2_combined_R1_val_1.fq.gz 02_trim/aefR-KB-2_combined/aefR-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/aefR-KB-2_combined/aefR-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/aefR-KB-2_combined/aefR-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/aefR-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/aefR-MM-1_combined/aefR-MM-1_combined_R1_val_1.fq.gz 02_trim/aefR-MM-1_combined/aefR-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/aefR-MM-1_combined/aefR-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/aefR-MM-1_combined/aefR-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/aefR-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/aefR-MM-2_combined/aefR-MM-2_combined_R1_val_1.fq.gz 02_trim/aefR-MM-2_combined/aefR-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/aefR-MM-2_combined/aefR-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/aefR-MM-2_combined/aefR-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/cvsS-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/cvsS-KB-1_combined/cvsS-KB-1_combined_R1_val_1.fq.gz 02_trim/cvsS-KB-1_combined/cvsS-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/cvsS-KB-1_combined/cvsS-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/cvsS-KB-1_combined/cvsS-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/cvsS-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/cvsS-KB-2_combined/cvsS-KB-2_combined_R1_val_1.fq.gz 02_trim/cvsS-KB-2_combined/cvsS-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/cvsS-KB-2_combined/cvsS-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/cvsS-KB-2_combined/cvsS-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/cvsS-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/cvsS-MM-1_combined/cvsS-MM-1_combined_R1_val_1.fq.gz 02_trim/cvsS-MM-1_combined/cvsS-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/cvsS-MM-1_combined/cvsS-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/cvsS-MM-1_combined/cvsS-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/cvsS-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/cvsS-MM-2_combined/cvsS-MM-2_combined_R1_val_1.fq.gz 02_trim/cvsS-MM-2_combined/cvsS-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/cvsS-MM-2_combined/cvsS-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/cvsS-MM-2_combined/cvsS-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/envZ-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/envZ-KB-1_combined/envZ-KB-1_combined_R1_val_1.fq.gz 02_trim/envZ-KB-1_combined/envZ-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/envZ-KB-1_combined/envZ-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/envZ-KB-1_combined/envZ-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/envZ-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/envZ-KB-2_combined/envZ-KB-2_combined_R1_val_1.fq.gz 02_trim/envZ-KB-2_combined/envZ-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/envZ-KB-2_combined/envZ-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/envZ-KB-2_combined/envZ-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/envZ-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/envZ-MM-1_combined/envZ-MM-1_combined_R1_val_1.fq.gz 02_trim/envZ-MM-1_combined/envZ-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/envZ-MM-1_combined/envZ-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/envZ-MM-1_combined/envZ-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/envZ-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/envZ-MM-2_combined/envZ-MM-2_combined_R1_val_1.fq.gz 02_trim/envZ-MM-2_combined/envZ-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/envZ-MM-2_combined/envZ-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/envZ-MM-2_combined/envZ-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gacA-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gacA-KB-1_combined/gacA-KB-1_combined_R1_val_1.fq.gz 02_trim/gacA-KB-1_combined/gacA-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gacA-KB-1_combined/gacA-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gacA-KB-1_combined/gacA-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gacA-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gacA-KB-2_combined/gacA-KB-2_combined_R1_val_1.fq.gz 02_trim/gacA-KB-2_combined/gacA-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gacA-KB-2_combined/gacA-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gacA-KB-2_combined/gacA-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gacA-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gacA-MM-1_combined/gacA-MM-1_combined_R1_val_1.fq.gz 02_trim/gacA-MM-1_combined/gacA-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gacA-MM-1_combined/gacA-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gacA-MM-1_combined/gacA-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gacA-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gacA-MM-2_combined/gacA-MM-2_combined_R1_val_1.fq.gz 02_trim/gacA-MM-2_combined/gacA-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gacA-MM-2_combined/gacA-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gacA-MM-2_combined/gacA-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gtrS-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gtrS-KB-1_combined/gtrS-KB-1_combined_R1_val_1.fq.gz 02_trim/gtrS-KB-1_combined/gtrS-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gtrS-KB-1_combined/gtrS-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gtrS-KB-1_combined/gtrS-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gtrS-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gtrS-KB-2_combined/gtrS-KB-2_combined_R1_val_1.fq.gz 02_trim/gtrS-KB-2_combined/gtrS-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gtrS-KB-2_combined/gtrS-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gtrS-KB-2_combined/gtrS-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gtrS-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gtrS-MM-1_combined/gtrS-MM-1_combined_R1_val_1.fq.gz 02_trim/gtrS-MM-1_combined/gtrS-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gtrS-MM-1_combined/gtrS-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gtrS-MM-1_combined/gtrS-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/gtrS-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/gtrS-MM-2_combined/gtrS-MM-2_combined_R1_val_1.fq.gz 02_trim/gtrS-MM-2_combined/gtrS-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/gtrS-MM-2_combined/gtrS-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/gtrS-MM-2_combined/gtrS-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ompR-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ompR-KB-1_combined/ompR-KB-1_combined_R1_val_1.fq.gz 02_trim/ompR-KB-1_combined/ompR-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ompR-KB-1_combined/ompR-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ompR-KB-1_combined/ompR-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ompR-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ompR-KB-2_combined/ompR-KB-2_combined_R1_val_1.fq.gz 02_trim/ompR-KB-2_combined/ompR-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ompR-KB-2_combined/ompR-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ompR-KB-2_combined/ompR-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ompR-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ompR-MM-1_combined/ompR-MM-1_combined_R1_val_1.fq.gz 02_trim/ompR-MM-1_combined/ompR-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ompR-MM-1_combined/ompR-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ompR-MM-1_combined/ompR-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ompR-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ompR-MM-2_combined/ompR-MM-2_combined_R1_val_1.fq.gz 02_trim/ompR-MM-2_combined/ompR-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ompR-MM-2_combined/ompR-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ompR-MM-2_combined/ompR-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ox-rpoN-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_R1_val_1.fq.gz 02_trim/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ox-rpoN-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_R1_val_1.fq.gz 02_trim/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ox-rpoN-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_R1_val_1.fq.gz 02_trim/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/ox-rpoN-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_R1_val_1.fq.gz 02_trim/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilR-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilR-KB-1_combined/pilR-KB-1_combined_R1_val_1.fq.gz 02_trim/pilR-KB-1_combined/pilR-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilR-KB-1_combined/pilR-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilR-KB-1_combined/pilR-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilR-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilR-KB-2_combined/pilR-KB-2_combined_R1_val_1.fq.gz 02_trim/pilR-KB-2_combined/pilR-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilR-KB-2_combined/pilR-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilR-KB-2_combined/pilR-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilR-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilR-MM-1_combined/pilR-MM-1_combined_R1_val_1.fq.gz 02_trim/pilR-MM-1_combined/pilR-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilR-MM-1_combined/pilR-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilR-MM-1_combined/pilR-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilR-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilR-MM-2_combined/pilR-MM-2_combined_R1_val_1.fq.gz 02_trim/pilR-MM-2_combined/pilR-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilR-MM-2_combined/pilR-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilR-MM-2_combined/pilR-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilS-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilS-KB-1_combined/pilS-KB-1_combined_R1_val_1.fq.gz 02_trim/pilS-KB-1_combined/pilS-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilS-KB-1_combined/pilS-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilS-KB-1_combined/pilS-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilS-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilS-KB-2_combined/pilS-KB-2_combined_R1_val_1.fq.gz 02_trim/pilS-KB-2_combined/pilS-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilS-KB-2_combined/pilS-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilS-KB-2_combined/pilS-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilS-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilS-MM-1_combined/pilS-MM-1_combined_R1_val_1.fq.gz 02_trim/pilS-MM-1_combined/pilS-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilS-MM-1_combined/pilS-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilS-MM-1_combined/pilS-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/pilS-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/pilS-MM-2_combined/pilS-MM-2_combined_R1_val_1.fq.gz 02_trim/pilS-MM-2_combined/pilS-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/pilS-MM-2_combined/pilS-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/pilS-MM-2_combined/pilS-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/psrA-KB-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/psrA-KB-1_combined/psrA-KB-1_combined_R1_val_1.fq.gz 02_trim/psrA-KB-1_combined/psrA-KB-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/psrA-KB-1_combined/psrA-KB-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/psrA-KB-1_combined/psrA-KB-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/psrA-KB-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/psrA-KB-2_combined/psrA-KB-2_combined_R1_val_1.fq.gz 02_trim/psrA-KB-2_combined/psrA-KB-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/psrA-KB-2_combined/psrA-KB-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/psrA-KB-2_combined/psrA-KB-2_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/psrA-MM-1_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/psrA-MM-1_combined/psrA-MM-1_combined_R1_val_1.fq.gz 02_trim/psrA-MM-1_combined/psrA-MM-1_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/psrA-MM-1_combined/psrA-MM-1_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/psrA-MM-1_combined/psrA-MM-1_combined_Aligned.sortedByCoord.out.bam\n",
      "mkdir 03_star/psrA-MM-2_combined\n",
      "STAR --runThreadN 20 --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --readFilesIn 02_trim/psrA-MM-2_combined/psrA-MM-2_combined_R1_val_1.fq.gz 02_trim/psrA-MM-2_combined/psrA-MM-2_combined_R2_val_2.fq.gz --outFileNamePrefix 03_star/psrA-MM-2_combined/psrA-MM-2_combined_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 --outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\n",
      "samtools index 03_star/psrA-MM-2_combined/psrA-MM-2_combined_Aligned.sortedByCoord.out.bam\n",
      "echo '************************************** mapping finished! **************************************'\n"
     ]
    }
   ],
   "source": [
    "# mapping by STAR    ref use genebank\n",
    "\n",
    "#STAR --runMode genomeGenerate --runThreadN 20 --sjdbGTFfeatureExon CDS --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 --genomeFastaFiles ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6/CP000058.1.fasta --sjdbGTFfile ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6/CP000058.1.gtf --sjdbOverhang 149 --genomeSAindexNbases 10  \n",
    "\n",
    "#ulimit -n 65535\n",
    "\n",
    "with open('03_star/03_star.sh', 'w') as f:\n",
    "    for i in sample:\n",
    "        print('mkdir 03_star/%s'%i)\n",
    "        f.write('mkdir 03_star/%s\\n'%i)\n",
    "        print('STAR --runThreadN %s --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 \\\n",
    "--readFilesIn 02_trim/%s/%s_R1_val_1.fq.gz 02_trim/%s/%s_R2_val_2.fq.gz \\\n",
    "--outFileNamePrefix 03_star/%s/%s_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 \\\n",
    "--outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat'%(thread,i,i,i,i,i,i))\n",
    "        f.write('STAR --runThreadN %s --genomeDir ~/data/star_index/PS_1448A_gb_CP000058.1_STAR2.6 \\\n",
    "--readFilesIn 02_trim/%s/%s_R1_val_1.fq.gz 02_trim/%s/%s_R2_val_2.fq.gz \\\n",
    "--outFileNamePrefix 03_star/%s/%s_ --outFilterType BySJout --outFilterMultimapNmax 1 --outFilterMismatchNmax 2 \\\n",
    "--outSAMtype BAM Unsorted SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --readFilesCommand zcat\\n'%(thread,i,i,i,i,i,i))\n",
    "        print('samtools index 03_star/%s/%s_Aligned.sortedByCoord.out.bam'%(i,i))\n",
    "        f.write('samtools index 03_star/%s/%s_Aligned.sortedByCoord.out.bam\\n'%(i,i))\n",
    "\n",
    "    print('echo \\'************************************** mapping finished! **************************************\\'')\n",
    "    f.write('echo \\'************************************** mapping finished! **************************************\\'')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir 03_tpm/1448A-WT-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/1448A-WT-KB-1_combined/1448A-WT-KB-1_combined_\n",
      "mkdir 03_tpm/1448A-WT-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/1448A-WT-KB-2_combined/1448A-WT-KB-2_combined_\n",
      "mkdir 03_tpm/1448A-WT-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/1448A-WT-MM-1_combined/1448A-WT-MM-1_combined_\n",
      "mkdir 03_tpm/1448A-WT-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/1448A-WT-MM-2_combined/1448A-WT-MM-2_combined_\n",
      "mkdir 03_tpm/PSPH-WT-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/PSPH-WT-KB-1_combined/PSPH-WT-KB-1_combined_\n",
      "mkdir 03_tpm/PSPH-WT-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/PSPH-WT-KB-2_combined/PSPH-WT-KB-2_combined_\n",
      "mkdir 03_tpm/PSPH-WT-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/PSPH-WT-MM-1_combined/PSPH-WT-MM-1_combined_\n",
      "mkdir 03_tpm/PSPH-WT-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/PSPH-WT-MM-2_combined/PSPH-WT-MM-2_combined_\n",
      "mkdir 03_tpm/S0858-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/S0858-KB-1_combined/S0858-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/S0858-KB-1_combined/S0858-KB-1_combined_\n",
      "mkdir 03_tpm/S0858-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/S0858-KB-2_combined/S0858-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/S0858-KB-2_combined/S0858-KB-2_combined_\n",
      "mkdir 03_tpm/S0858-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/S0858-MM-1_combined/S0858-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/S0858-MM-1_combined/S0858-MM-1_combined_\n",
      "mkdir 03_tpm/S0858-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/S0858-MM-2_combined/S0858-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/S0858-MM-2_combined/S0858-MM-2_combined_\n",
      "mkdir 03_tpm/Vfr-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/Vfr-KB-1_combined/Vfr-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/Vfr-KB-1_combined/Vfr-KB-1_combined_\n",
      "mkdir 03_tpm/Vfr-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/Vfr-KB-2_combined/Vfr-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/Vfr-KB-2_combined/Vfr-KB-2_combined_\n",
      "mkdir 03_tpm/Vfr-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/Vfr-MM-1_combined/Vfr-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/Vfr-MM-1_combined/Vfr-MM-1_combined_\n",
      "mkdir 03_tpm/Vfr-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/Vfr-MM-2_combined/Vfr-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/Vfr-MM-2_combined/Vfr-MM-2_combined_\n",
      "mkdir 03_tpm/aefR-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/aefR-KB-1_combined/aefR-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/aefR-KB-1_combined/aefR-KB-1_combined_\n",
      "mkdir 03_tpm/aefR-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/aefR-KB-2_combined/aefR-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/aefR-KB-2_combined/aefR-KB-2_combined_\n",
      "mkdir 03_tpm/aefR-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/aefR-MM-1_combined/aefR-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/aefR-MM-1_combined/aefR-MM-1_combined_\n",
      "mkdir 03_tpm/aefR-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/aefR-MM-2_combined/aefR-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/aefR-MM-2_combined/aefR-MM-2_combined_\n",
      "mkdir 03_tpm/cvsS-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/cvsS-KB-1_combined/cvsS-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/cvsS-KB-1_combined/cvsS-KB-1_combined_\n",
      "mkdir 03_tpm/cvsS-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/cvsS-KB-2_combined/cvsS-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/cvsS-KB-2_combined/cvsS-KB-2_combined_\n",
      "mkdir 03_tpm/cvsS-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/cvsS-MM-1_combined/cvsS-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/cvsS-MM-1_combined/cvsS-MM-1_combined_\n",
      "mkdir 03_tpm/cvsS-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/cvsS-MM-2_combined/cvsS-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/cvsS-MM-2_combined/cvsS-MM-2_combined_\n",
      "mkdir 03_tpm/envZ-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/envZ-KB-1_combined/envZ-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/envZ-KB-1_combined/envZ-KB-1_combined_\n",
      "mkdir 03_tpm/envZ-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/envZ-KB-2_combined/envZ-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/envZ-KB-2_combined/envZ-KB-2_combined_\n",
      "mkdir 03_tpm/envZ-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/envZ-MM-1_combined/envZ-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/envZ-MM-1_combined/envZ-MM-1_combined_\n",
      "mkdir 03_tpm/envZ-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/envZ-MM-2_combined/envZ-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/envZ-MM-2_combined/envZ-MM-2_combined_\n",
      "mkdir 03_tpm/gacA-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gacA-KB-1_combined/gacA-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gacA-KB-1_combined/gacA-KB-1_combined_\n",
      "mkdir 03_tpm/gacA-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gacA-KB-2_combined/gacA-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gacA-KB-2_combined/gacA-KB-2_combined_\n",
      "mkdir 03_tpm/gacA-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gacA-MM-1_combined/gacA-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gacA-MM-1_combined/gacA-MM-1_combined_\n",
      "mkdir 03_tpm/gacA-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gacA-MM-2_combined/gacA-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gacA-MM-2_combined/gacA-MM-2_combined_\n",
      "mkdir 03_tpm/gtrS-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gtrS-KB-1_combined/gtrS-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gtrS-KB-1_combined/gtrS-KB-1_combined_\n",
      "mkdir 03_tpm/gtrS-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gtrS-KB-2_combined/gtrS-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gtrS-KB-2_combined/gtrS-KB-2_combined_\n",
      "mkdir 03_tpm/gtrS-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gtrS-MM-1_combined/gtrS-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gtrS-MM-1_combined/gtrS-MM-1_combined_\n",
      "mkdir 03_tpm/gtrS-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/gtrS-MM-2_combined/gtrS-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/gtrS-MM-2_combined/gtrS-MM-2_combined_\n",
      "mkdir 03_tpm/ompR-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ompR-KB-1_combined/ompR-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ompR-KB-1_combined/ompR-KB-1_combined_\n",
      "mkdir 03_tpm/ompR-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ompR-KB-2_combined/ompR-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ompR-KB-2_combined/ompR-KB-2_combined_\n",
      "mkdir 03_tpm/ompR-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ompR-MM-1_combined/ompR-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ompR-MM-1_combined/ompR-MM-1_combined_\n",
      "mkdir 03_tpm/ompR-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ompR-MM-2_combined/ompR-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ompR-MM-2_combined/ompR-MM-2_combined_\n",
      "mkdir 03_tpm/ox-rpoN-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ox-rpoN-KB-1_combined/ox-rpoN-KB-1_combined_\n",
      "mkdir 03_tpm/ox-rpoN-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ox-rpoN-KB-2_combined/ox-rpoN-KB-2_combined_\n",
      "mkdir 03_tpm/ox-rpoN-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ox-rpoN-MM-1_combined/ox-rpoN-MM-1_combined_\n",
      "mkdir 03_tpm/ox-rpoN-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/ox-rpoN-MM-2_combined/ox-rpoN-MM-2_combined_\n",
      "mkdir 03_tpm/pilR-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilR-KB-1_combined/pilR-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilR-KB-1_combined/pilR-KB-1_combined_\n",
      "mkdir 03_tpm/pilR-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilR-KB-2_combined/pilR-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilR-KB-2_combined/pilR-KB-2_combined_\n",
      "mkdir 03_tpm/pilR-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilR-MM-1_combined/pilR-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilR-MM-1_combined/pilR-MM-1_combined_\n",
      "mkdir 03_tpm/pilR-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilR-MM-2_combined/pilR-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilR-MM-2_combined/pilR-MM-2_combined_\n",
      "mkdir 03_tpm/pilS-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilS-KB-1_combined/pilS-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilS-KB-1_combined/pilS-KB-1_combined_\n",
      "mkdir 03_tpm/pilS-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilS-KB-2_combined/pilS-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilS-KB-2_combined/pilS-KB-2_combined_\n",
      "mkdir 03_tpm/pilS-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilS-MM-1_combined/pilS-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilS-MM-1_combined/pilS-MM-1_combined_\n",
      "mkdir 03_tpm/pilS-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/pilS-MM-2_combined/pilS-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/pilS-MM-2_combined/pilS-MM-2_combined_\n",
      "mkdir 03_tpm/psrA-KB-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/psrA-KB-1_combined/psrA-KB-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/psrA-KB-1_combined/psrA-KB-1_combined_\n",
      "mkdir 03_tpm/psrA-KB-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/psrA-KB-2_combined/psrA-KB-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/psrA-KB-2_combined/psrA-KB-2_combined_\n",
      "mkdir 03_tpm/psrA-MM-1_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/psrA-MM-1_combined/psrA-MM-1_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/psrA-MM-1_combined/psrA-MM-1_combined_\n",
      "mkdir 03_tpm/psrA-MM-2_combined\n",
      "rsem-calculate-expression --paired-end -p 20 --bam 03_star/psrA-MM-2_combined/psrA-MM-2_combined_Aligned.toTranscriptome.out.bam ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/psrA-MM-2_combined/psrA-MM-2_combined_\n",
      "echo '************************************** mapping finished! **************************************'\n"
     ]
    }
   ],
   "source": [
    "# calculate tpm\n",
    "\n",
    "# rsem-prepare-reference --gtf ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/CP000058.1.gtf --star -p 20 ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/CP000058.1.fasta ~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A\n",
    "\n",
    "with open('03_tpm/03_tpm.sh', 'w') as f:\n",
    "    for i in sample:\n",
    "        print('mkdir 03_tpm/%s'%i)\n",
    "        f.write('mkdir 03_tpm/%s\\n'%i)\n",
    "        print('rsem-calculate-expression --paired-end -p %s --bam 03_star/%s/%s_Aligned.toTranscriptome.out.bam \\\n",
    "~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/%s/%s_'%(thread,i,i,i,i))\n",
    "        f.write('rsem-calculate-expression --paired-end -p %s --bam 03_star/%s/%s_Aligned.toTranscriptome.out.bam \\\n",
    "~/data/rsem_star_index/PS_1448A_gb_CP000058.1/PS_1448A 03_tpm/%s/%s_\\n'%(thread,i,i,i,i))\n",
    "\n",
    "    print('echo \\'************************************** mapping finished! **************************************\\'')\n",
    "    f.write('echo \\'************************************** mapping finished! **************************************\\'')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Attention\n",
    "\n",
    "When build STAR index, only CDS regions are considered (4985),\n",
    "When build rsem index, only exon regions are considered (80), which is tRNA,\n",
    "Here we change exon into CDS in the gtf file, so we have 4985+80 regions in the rsem index."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
